Sources/SwiftGRPCNIO/CallHandlers/ServerStreamingCallHandler.swift

```diff
@@ -26,12 +26,10 @@ public class ServerStreamingCallHandler<RequestMessage: Message, ResponseMessage
   }
   
   
-  public override func processMessage(_ message: RequestMessage) {
+  public override func processMessage(_ message: RequestMessage) throws {
     guard let eventObserver = self.eventObserver,
       let context = self.context else {
-      //! FIXME: Better handle this error?
-      print("multiple messages received on unary call")
-      return
+        throw GRPCStatus(code: .unimplemented, message: "multiple messages received on unary call")
```

Comment:
Is `.unimplemented` appropriate here?

Also, how about having a dedicated enum of all possible connection errors generated by gRPC (e.g. "attempted to write more than one status", "multiple messages received on unary call", `requestProtoParseError`, "unimplemented", `responseProtoSerializationError`, `unsupportedCompression` etc.) which conforms to `GRPCStatusTransformable`, instead of using `GRPCStatus` directly? The reasoning would be that this reduces the variety of errors the error delegate would need to expect/care about, but still lets us return the appropriate status to the client (thanks to the enum's conformance to `GRPCStatusTransformable`). Another advantage would be that we don't need to update this error string in two places, for example. (It is also used in `UnaryCallHandler.swift`.)

Reply:
The documentation (https://github.com/grpc/grpc/blob/master/doc/statuscodes.md) suggests unimplemented. See:

> "Request cardinality violation (method requires exactly one request but client sent some other number of requests)"

--- 

Re: the dedicated enum of errors; I like that idea a lot. Another advantage of having them defined in one place is that it's a clear list of states to test for.




>

Reply:
Thanks for the elaboration, and sorry for missing that!

And glad you like the idea of the error enum. Note that my enumeration might not be exhaustive; we might want to search the repo for all `GRPCStatus(` calls to figure out which ones we should add to that enum (probably only the ones that can possibly be sent to the error delegate, though).